src/openms/source/ANALYSIS/OPENSWATH/MRMIonSeries.cpp

```diff
@@ -257,10 +257,24 @@ namespace OpenMS
 
   void MRMIonSeries::annotateTransition(ReactionMonitoringTransition& tr, const TargetedExperiment::Peptide peptide, const double precursor_mz_threshold, double product_mz_threshold, bool enable_reannotation, std::vector<String> fragment_types, std::vector<size_t> fragment_charges, bool enable_specific_losses, bool enable_unspecific_losses, int round_decPow)
   {
-    CVTermList interpretation;
+    OPENMS_PRECONDITION(peptide.hasCharge(), "Cannot annotate transition without a peptide charge state")
+    // TODO: we should not have transitions without charge states
+    // OPENMS_PRECONDITION(tr.isProductChargeStateSet(), "Cannot annotate transition without a charge state")
```

Comment:
@grosenberger  can you comment on this? It seems that sometimes the product charge state is not set and this precondition fails. This is not ideal as we then have to make an assumption about the fragment charge state here. Before my edit, this charge would have been `-1` which is pretty bad (e.g. it would subtract a hydrogen atom instead of adding one...)

Also, what is the parameter `fragment_charges` supposed to provide, does it contain potential charge states?